lib/openhab/core/emulate_hash.rb

```diff
@@ -0,0 +1,233 @@
+# frozen_string_literal: true
+
+require "forwardable"
+
+module OpenHAB
+  module Core
+    #
+    # Make an openHAB class acts like Ruby Hash.
+    #
+    module EmulateHash
+      include Enumerable
+      extend Forwardable
+
+      # The implementing class must implement #to_hash
+      def_delegators :to_hash,
```

Comment:
I'm concerned about delegating all of these to `to_hash`, which always returns a _Ruby_ Hash, and thus a temporary object for all of these methods that are already implemented by java.util.Map (or at least by JRuby's wrapper of java.util.Map). That's fine for `MetadataHash`, which doesn't have an actual java.util.Map, but Metadata::Hash and Configuration do. So I would suggest that the required interface be the including class provides `to_map` (as a private method), and then delegate all the methods that exist on Map (including `to_hash`) to `to_map`, and methods that don't to `to_hash`. On Configuration and MetadataHash, `to_map` can just return what their `to_hash` methods currently return (`Hash#to_hash` returns self, so it won't cause extra copies).